fpdf/html.py

```diff
@@ -281,7 +280,8 @@ def __init__(
         tag_indents=None,
         tag_styles=None,
         list_vertical_margin=None,
-        **_,
+        heading_above=0.2,
```

Comment:
By the way, in terms of adopting the best future-proof / generic / handy end-user API,
maybe we should change the type of `tag_styles` to be an array of `TitleStyle`,
and convert those default values of `heading_above` / `heading_below`,
into `t_margin=0.2` & `b_margin=0.2` properties of the new default value of `tag_styles` for `h1`...`h6` headings.

Do you agree @gmischler?

Reply:
There seems to be a bit of an inflation of styling classes... I haven't really looked into that topic myself.

It is extremely inconvenient that this one is called "TitleStyle". I understand that it was originally created to format section titles, but it should not have been given such a single-purpose name. Something like "TextStyle" would be much more appropriate.

In our case here, it would indeed be interesting to use in `tag_styles`. Given that it is derived from `FontFace`, that won't even break anything. We even already have a `combine()` method. Can this be used to pass the values from a user-supplied `FontFace` into an internal `TextStyle`?

I also think that `tag_indents` should really be part of `TextStyle`. It's really weird that `tag_styles` and `tag_indents` were introduced parallel to each other (the former already has `l_margin`). Or do we want to further increase the styling class inflation and create another subclass `ParagraphStyle`?

It becomes increasingly clear that all those styling features were slapped together without first trying to come up with a real concept of which role each one is supposed to play. It might be good to take the time now to figure out something a bit more systematic.

Reply:
> It is extremely inconvenient that this one is called "TitleStyle". I understand that it was originally created to format section titles, but it should not have been given such a single-purpose name. Something like "TextStyle" would be much more appropriate.

Alright, I'm OK to perform this class renaming, while preserving `TitleStyle` as an alias for backward compatibility.

> In our case here, it would indeed be interesting to use in `tag_styles`. Given that it is derived from `FontFace`, that won't even break anything. We even already have a `combine()` method. Can this be used to pass the values from a user-supplied `FontFace` into an internal `TextStyle`?

Yes, I think I can do that 👍

> I also think that `tag_indents` should really be part of `TextStyle`.

If I understdood you correctly, then I agree.

> It's really weird that `tag_styles` and `tag_indents` were introduced parallel to each other (the former already has `l_margin`). Or do we want to further increase the styling class inflation and create another subclass `ParagraphStyle`?

No, I agree that we should avoid styling class inflation.

Currently `tag_styles` are currently instances of `FontFace`.
I'm going to convert them to instances of `TextStyle` and deprecate `tag_indents`.